godot/RandomWalker/Actors/Enemy.gd

```diff
@@ -1,15 +1,30 @@
 extends "Actor.gd"
 
 
-onready var visibility_enabler : VisibilityEnabler2D = $VisibilityEnabler2D
-
+@onready var visibility_enabler : VisibleOnScreenEnabler2D = $VisibleOnScreenEnabler2D
+@onready var inital_speed : float = 300.0
+@onready var _direction : float = 1.0
 
 func _ready() -> void:
```

Comment:
In simple cases like this we prefer to connect directly to the built-in functions instead of creating new ones just for a call. We also prefer connecting in code because it's easier to keep track of all the connections:
```
	visibility_enabler.screen_entered.connect(set_physics_process.bind(true))
	visibility_enabler.screen_exited.connect(set_physics_process.bind(false))
```

We can then remove `_on_visible_on_screen_enabler_2d_*()` functions